### PR TITLE
Fix: Shows metadata in treepicker when nodes contain ancestors data

### PR DESCRIPTION
**Pull request by mdotwills**
**Background**
Nodes loaded from the api as already selected describe their path in terms of ancestry with the `ancestors` key.

**Changes**
- Checks both `node.path` and `node.ancestory` when rendering metadata.

**Trello**
https://trello.com/c/Xbs2mGJ0/5656-multiple-segments-description-missing-for-selected-cities-and-states

### CODE REVIEW

File: src/components/adslotUi/TreePickerNodeComponent.jsx [hunk 1]
```diff
@@ -34,9 +34,12 @@ const TreePickerNodeComponent = ({
...
   valueFormatter,
 }) => {
-  const pathElement = !_.isEmpty(node.path) ?
+
+  const pathElement = !(_.isEmpty(node.path) && _.isEmpty(node.ancestors)) ?
     <span className={`${baseClass}-path`}>
-      {_(node.path).map('label').clone().reverse().join(', ')}
+      { !_.isEmpty(node.path) ?
+        _(node.path).map('label').clone().reverse().join(', ') :
+        _(node.ancestors).map('label').clone().join(', ') }
```
Comment (OndrejCholeva) [inline src/components/adslotUi/TreePickerNodeComponent.jsx]: I think the clone is extra here, in fact `_.map(node.ancestors, 'label').join(', ')` should be just fine.

File: src/components/adslotUi/TreePickerNodeComponent.jsx [hunk 2]
```diff
@@ -34,9 +34,12 @@ const TreePickerNodeComponent = ({
   selected,
   valueFormatter,
 }) => {
-  const pathElement = !_.isEmpty(node.path) ?
+
+  const pathElement = !(_.isEmpty(node.path) && _.isEmpty(node.ancestors)) ?
```
Comment (patrick-ivers) [inline src/components/adslotUi/TreePickerNodeComponent.jsx]: Could you change `!(_.isEmpty(node.path) && _.isEmpty(node.ancestors))` to `_.some([node.path, node.ancestors])` ?

File: src/components/adslotUi/TreePickerNodeComponent.jsx [hunk 3]
```diff
@@ -34,9 +34,12 @@ const TreePickerNodeComponent = ({
...
     <span className={`${baseClass}-path`}>
-      {_(node.path).map('label').clone().reverse().join(', ')}
+      { !_.isEmpty(node.path) ?
+        _(node.path).map('label').clone().reverse().join(', ') :
```
Comment (patrick-ivers) [inline src/components/adslotUi/TreePickerNodeComponent.jsx]: Is `.clone()` necessary?

Comment (OndrejCholeva) [inline src/components/adslotUi/TreePickerNodeComponent.jsx]: clone is only necessary when operating on original array as reverse mutates it. Given that map is creating new array, it is safe to skip it. both reverse and join are native so there no need to even initiate lodash chain:
```
_.map(node.path, 'label').reverse().join(', '):
_.map(node.path, 'label').join(', ')
```

Comment (mdotwills) [inline src/components/adslotUi/TreePickerNodeComponent.jsx]: @OndrejCholeva, seems the `clone()` is needed after all (unit tests fail) 😞 Could just be with the way they were setup, but I'm not going to change them for sake of removing a `clone` that was already present.

File: src/components/adslotUi/TreePickerNodeComponent.jsx (hunk 2 above)
Comment (mdotwills) [inline src/components/adslotUi/TreePickerNodeComponent.jsx]: @patrick-ivers does not work, even when specifying the `predicate` arg.

**Comment by mdotwills** [comment]
Merge?

File: src/components/adslotUi/TreePickerNodeComponent.jsx (hunk 2 above)
Comment (OndrejCholeva) [inline src/components/adslotUi/TreePickerNodeComponent.jsx]: `!_.every([node.path, node.ancestors], _.isEmpty)` would work, but anyway...

File: src/components/adslotUi/TreePickerNodeComponent.jsx (hunk 3 above)
Comment (omgaz) [inline src/components/adslotUi/TreePickerNodeComponent.jsx]: The intent of this line I think get's lost in lodash. Can you maybe take another look or at least do this outside of the JSX wrapped in an explanatory function.

File: test/components/adslotUi/TreePickerNodeComponentTest.jsx [hunk 4]
```diff
@@ -7,7 +7,7 @@ import { GridCell, GridRow } from 'alexandria-adslot';
 import { shallow, mount } from 'enzyme';
 
 describe('TreePickerNodeComponent', () => {
-  const { cbrNode, actNode, itemType, nodeRenderer } = TreePickerMocks;
+  const { cbrNode, cbrNodeAlreadySelected, actNode, itemType, nodeRenderer } = TreePickerMocks;
```
Comment (omgaz) [inline test/components/adslotUi/TreePickerNodeComponentTest.jsx]: What was cbr again?

Comment (OndrejCholeva) [inline test/components/adslotUi/TreePickerNodeComponentTest.jsx]: **C**an**b**e**r**ra?

**Pull request merged**